src/nvim/version.c

```diff
@@ -2715,8 +2715,10 @@ void list_version(void)
 /// Whether it still is not too late to show an intro message
 bool may_show_intro(void)
 {
-  return (buf_is_empty(firstwin->w_buffer)
-          && (firstwin->w_buffer->b_fname == NULL)
+  return (buf_is_empty(curbuf)
+          && (curbuf->b_fname == NULL)
+          && (curbuf->handle == 1)
+          && (curwin->handle == LOWEST_WIN_ID)
```

Comment:
This will show intro if the window 1000 was split, then made into a floating window using `nvim_win_set_config`.

Reply:
Realized that adding `&& (!curwin->w_floating)` is working here and is aligned with the description of conditions ("...single non-floating...").

*Now* I hope there is no (reasonable) edge cases?